SIPS/sip-16.md

```diff
@@ -0,0 +1,131 @@
+
+## Motivation
+
+One of the most difficult problems blockchain wallets solve for their users is "signature comprehension," i.e. making cryptographic signature inputs intelligible to the user.
+A signature in the wrong hands can give an attacker the ability to steal user assets. 
+A single wallet may not be able to provide all relevant information to any given user for any given signature request.
+To alleviate this problem, this SIP aims to expand the kinds of information MetaMask provides to a user before signing a message.
+
+The current Snaps API in the MetaMask extension already has a "transaction insights" feature that allows Snaps to decode transactions and provide insights to users. 
+These transaction insights can also specify a transaction severity level to provide extra friction in the transaction confirmation. 
+To expand on this feature, this SIP allows the community to build Snaps that provide arbitrary "insights" into signatures in a similar manner.
+These insights can then be displayed in the MetaMask UI alongside any information provided by MetaMask itself.
+Signature insight Snaps can also return a `severity` key to indicate the severity level of the content being returned. 
+This key uses the same `SeverityLevel` enum used by transaction insight Snaps. 
+
+## Specification
+
+> Formal specifications are written in Typescript. Usage of `CAIP-N` specifications, where `N` is a number, are references to [Chain Agnostic Improvement Proposals](https://github.com/ChainAgnostic/CAIPs).
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+> This section is non-normative, and merely recapitulates some definitions from [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md).
+
+- `ChainId` - a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) string.
+  It identifies a specific chain among all blockchains recognized by the CAIP standards. The ChainId refers to the network that the signing account is connected to. It is not necessarily the network where the signed message will be broadcast. 
+  - `ChainId` consists of a `Namespace` and a `Reference`
+    - `Namespace` - a class of similar blockchains. For example EVM-based blockchains.
+    - `Reference` - a way to identify a concrete chain inside a `Namespace`. For example Ethereum Mainnet or one of its test networks.
+- `Method` - a string which can be any of the following 5 [signing methods](https://docs.metamask.io/wallet/concepts/signing-methods/) supported by the MetaMask wallet API. The signing method is required by MetaMask for the dapp to make a signature request. It is not part of the signature payload.
```

Comment:
It might be worth it to mention that this list would be expanded with custom signature types in the future.

Reply:
Added a sentence in latest commit